### DISCUSSION TITLE
How make setter and getter on adonisjs 6

### DISCUSSION THREAD

**Discussion by adityadarma** [adonisjs/core#4594] [Help]
Hi team

i will make setter and getter like laravel (getAttribute and setAttribute), i will make encryption database when condition save data will encrypt and get data will decrypt
i have try with decorator afterFind and afterFetch, but cant success

**Comment by RomainLanz** [comment]
Hey @adityadarma! 👋🏻 

I am not sure what you mean, but if you want to transform the value before and after fetching it from the database, you can use the `consume` and `prepare` hook.

Example:

```ts
@column({
  serializeAs: null,
  prepare: (value: string | null) => (value ? encryption.encrypt(value) : null),
  consume: (value: string | null) => (value ? encryption.decrypt(value) : null),
})
public shopifyToken: string | null~
```
